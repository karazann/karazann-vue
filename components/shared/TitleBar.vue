<template lang="pug">
    section
        .container
            .row
                .col
                    h2 {{title}}
                    span(v-if="subtitle") {{subtitle}}
</template>

<script>
	export default {
		name: 'title-bar',
		props: {
			title: {
				type: String,
				required: true
			},
			subtitle: String
		}
	}
</script>

<style lang="scss" scoped>
	section {
        padding-top: 55px;
        padding-bottom: 30px;
    }
    .row {
        display: flex;
        flex: 0 0 100%;
        max-width: 100%;
    }
    .col {
        @include make-col-ready;
        @include make-col(12);
    }
	h2 {
		font-size: 30px;
		line-height: 40px;
		margin: 1px 0 3px;
		color: #334;
		font-weight: 500;
		margin: 0;
		padding: 0;
	}

	span {
		font-size: 20px;
		color: #889;
		margin-bottom: 3px;
		margin-top: 3px;
		display: block;
	}
</style>



