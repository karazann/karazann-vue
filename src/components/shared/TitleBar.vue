<template lang="pug">
    section
        .container
            .row
                .col
                    h1 {{title}}
                    p.lg(v-if="subtitle") {{subtitle}}
</template>

<script lang="ts">
    import Vue from 'vue'

    export default Vue.extend({
        name: 'title-bar',
        props: {
            title: {
                type: String,
                required: true
            },
            subtitle: String
        }
    })
</script>

<style lang="scss" scoped>
    section {
        padding-top: 55px;
        padding-bottom: 30px;
    }

    .row {
        display: flex;
        flex: 0 0 100%;
        max-width: 100%;
    }

    .col {
        @include make-col-ready;
        @include make-col(12);
    }

    h1 {
        line-height: 40px;
        margin: 1px 0 3px;
        margin: 0;
        padding: 0;
    }

    p {
        margin-bottom: 3px;
        margin-top: 3px;
        display: block;
    }
</style>
