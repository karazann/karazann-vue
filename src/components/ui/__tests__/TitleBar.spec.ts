import { shallowMount } from '@vue/test-utils'
import TitleBar from '../TitleBar.vue'

describe('TitleBar', () => {
    const build = (props: Record<string, any>) => {
        const wrapper = shallowMount(TitleBar, {
            propsData: props
        })

        return {
            wrapper
        }
    }

    it('renders the component', () => {
        // arrange
        const { wrapper } = build({
            title: 'Title',
            subtitle: ''
        })

        // assert
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders title props properly', () => {
        // arrange
        const { wrapper } = build({
            title: 'Title',
            subtitle: 'Subtitle'
        })

        // assert
        expect(wrapper.html()).toMatchSnapshot()
    })
})
