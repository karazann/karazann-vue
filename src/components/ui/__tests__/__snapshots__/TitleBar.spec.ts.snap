// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar renders the component 1`] = `
"<section>
  <div class=\\"container\\">
    <div class=\\"row\\">
      <div class=\\"col\\">
        <h1>Title</h1>
        <!---->
      </div>
    </div>
  </div>
</section>"
`;

exports[`TitleBar renders title props properly 1`] = `
"<section>
  <div class=\\"container\\">
    <div class=\\"row\\">
      <div class=\\"col\\">
        <h1>Title</h1>
        <p class=\\"lg\\">Subtitle</p>
      </div>
    </div>
  </div>
</section>"
`;
